lib/thoth-init.sh

```diff
@@ -0,0 +1,239 @@
+# shell doesn't run us again.
+#
+unset BASH_ENV
+
+thoth_fatal()
+{
+	echo thoth: "$*" 1>&2
+	exit 1
+}
+
+#
+# Unlike `thoth load` alone, this updates the index, the Manta file, and the
+# local copy ./info.json.
+#
+thoth_load()
+{
+	local update=$1
+
+	if [[ -f "$update" ]]; then
+		local tmpfile=$THOTH_TMPDIR/thoth.out.$$
+
+		cat $THOTH_INFO $update | json --deep-merge >$tmpfile
+		mv $tmpfile $THOTH_INFO
+	fi
+
+	mput -qf $THOTH_INFO $THOTH_INFO_OBJECT
+	$THOTH load $THOTH_INFO
+}
+
+#
+# "sysprops" are those at the root, not under .properties
+#
+thoth_set_sys()
+{
+	local prop=$1
+	local propfile=$THOTH_TMPDIR/thoth.prop.$$
+
+	if [[ "$#" -lt 1 ]]; then
+		thoth_fatal "failed to specify property"
+	fi
+
+	if [[ "$#" -eq 2 ]]; then
+		local propval="$2"
+	else
+		local propval=`cat | sed 's/\\\\/\\\\\\\\/g' | \
+		    sed 's/"/\\\\"/g' | sed ':a;N;$!ba;s/\\n/\\\\n/g'`
+	fi
+
+	echo "{ \"$prop\": \"$propval\" }" >$propfile
+	thoth_load $propfile
+}
+
+thoth_set()
+{
+	local prop=$1
+	local propfile=$THOTH_TMPDIR/thoth.prop.$$
+
+	if [[ "$#" -lt 1 ]]; then
+		thoth_fatal "failed to specify property"
+	fi
+
+	if [[ "$#" -eq 2 ]]; then
+		local propval="$2"
+	else
+		local propval=`cat | sed 's/\\\\/\\\\\\\\/g' | \
+		    sed 's/"/\\\\"/g' | sed ':a;N;$!ba;s/\\n/\\\\n/g'`
+	fi
+
+	echo "{ \"properties\": { \"$prop\": \"$propval\" } }" >$propfile
+	thoth_load $propfile
+}
+
+#
+# "sysprops" are those at the root, not under .properties
+#
+thoth_unset_sys()
+{
+	local tmpfile=$THOTH_TMPDIR/thoth.out.$$
+
+	if [[ "$#" -lt 1 ]]; then
+		thoth_fatal "failed to specify property"
+	fi
+
+	cat $THOTH_INFO | json -e "this.$1=undefined" >$tmpfile
+	mv $tmpfile $THOTH_INFO
+	thoth_load $propfile
+}
+
+thoth_unset()
+{
+	local tmpfile=$THOTH_TMPDIR/thoth.out.$$
+
+	if [[ "$#" -lt 1 ]]; then
+		thoth_fatal "failed to specify property"
+	fi
+
+	cat $THOTH_INFO | json -e "this.properties.$1=undefined" >$tmpfile
+	mv $tmpfile $THOTH_INFO
+	thoth_load $propfile
+}
+
+thoth_ticket()
+{
+	thoth_set_sys ticket $*
+}
+
+thoth_unticket()
+{
+	thoth_unset_sys ticket
+}
+
+thoth_analyze()
+{
+	. $THOTH_ANALYZER
+}
+
+export THOTH_TMPDIR=$(pwd)
+export THOTH_DUMP=$MANTA_INPUT_FILE
+export THOTH_NAME=$(basename $(dirname $MANTA_INPUT_FILE))
+export THOTH_INFO=$THOTH_TMPDIR/info.json
+export THOTH_DIR=$(dirname $MANTA_INPUT_OBJECT)
+export THOTH_INFO_OBJECT=$THOTH_DIR/info.json
+
+#
+# As `thoth load` only updates the index, it is the info of record, but we fall
+# back to the Manta file if needed.
+#
+if [[ ! -f $THOTH_INFO ]]; then
+	$THOTH info $THOTH_NAME >$THOTH_INFO 2>/dev/null ||
+	    mget -q $THOTH_INFO_OBJECT >$THOTH_INFO 2>/dev/null
+fi
+
+export THOTH_TYPE=$(cat $THOTH_INFO | json type)
+
+export PS1="thoth@$THOTH_NAME $ "
+export DTRACE_DOF_INIT_DISABLE=1
+
+if [[ $(cat $THOTH_INFO | json cmd) == "node" ]]; then
+	FILE_STR="$(file $THOTH_DUMP)"
+	MDB_V8=mdb_v8_ia32.so
+	export MDB_V8_DIR=$THOTH_TMPDIR/mdb
+
+	function jmget() {
```

Comment:
Why do we use curl -k in some places in this script, and normal mget elsewhere?

Reply:
No idea, but this script isn't new.

Reply:
This is two different Mantas: we use curl for us-east so we don't require a suitable key setup, and the other one is where thoth is actually hosted.